routers/web/repo/actions/badge.go

```diff
@@ -19,13 +19,33 @@ import (
 func GetWorkflowBadge(ctx *context.Context) {
 	workflowFile := ctx.PathParam("workflow_name")
 	branch := ctx.Req.URL.Query().Get("branch")
-	if branch == "" {
+	tag := ctx.Req.URL.Query().Get("tag")
+	useLatestTag := ctx.Req.URL.Query().Has("latest_tag")
+	var ref string
+	switch {
+	case useLatestTag:
+		tags, _, err := ctx.Repo.GitRepo.GetTagInfos(0, 1)
+		if err != nil {
+			ctx.ServerError("GetTagInfos", err)
+			return
+		}
+		if len(tags) != 0 {
+			tag = tags[0].Name
+		} else {
+			tag = "" // return empty result on no tag
+		}
+		ref = fmt.Sprintf("refs/tags/%s", tag)
+	case tag != "":
+		ref = fmt.Sprintf("refs/tags/%s", tag)
+	case branch != "":
+		ref = fmt.Sprintf("refs/heads/%s", branch)
+	default:
 		branch = ctx.Repo.Repository.DefaultBranch
```

Comment:
This assign statement is unnecessary.

Reply:
Should I

```
if useLatestTag {
	tag = "xxx"
}
switch {
  // ...
}
```
or add a `fallthrough`?

Reply:
I think it's also OK we handle useLatestTag before switch.